operations/iuf/IUF.md

```diff
@@ -0,0 +1,509 @@
+  -s STATE_DIR, --state-dir STATE_DIR
+                        A directory used to store the current state of stages. Defaults to [base-dir]/state.
+  -m MEDIA_DIR, --media-dir MEDIA_DIR
+                        Location of installation media you would like to install. Defaults to
+                        $RBD_BASE_DIR/{activity_session}
+  -mh MEDIA_HOST, --media-host MEDIA_HOST
+                        Host to extract the media onto. Defaults to ncn-m001
+  --log-dir LOG_DIR     Location used to store log files. Defaults to [base-dir]/log
+  -l {CRITICAL,ERROR,WARNING,INFO,DEBUG,TRACE}, --level {CRITICAL,ERROR,WARNING,INFO,DEBUG,TRACE}
+                        Set the debug level to the console
+  -v, --verbose         Generates more verbose messages
+
+subcommands:
+  {run,activity,list-stages|ls,list-products|resume|restart|abort}
+```
+
+### Input File
+
+As described in the [Output and Log Files](#output-and-log-files) section, the `-i INPUT_FILE` argument can be used to read `iuf` arguments and values from a YAML input file. Both global and subcommand-specific arguments can be
+specified in the input file. If an input file is used in addition to `iuf` arguments, the `iuf` arguments take precedence. The name of an entries in the input file corresponds to the long form name of the `iuf` argument with
+hyphens replaced by underscores.
+
+The following in an example of an `iuf` input file:
+
+```yaml
+global:
+    activity_session: joe-install-20230107
+    base_dir: /home/admin1/hpc-csm-software-recipe-22.11.7
+    state_dir: /home/admin1/hpc-csm-software-recipe-22.11.7/state
+    media_dir: /home/admin1/hpc-csm-software-recipe-22.11.7/media
+list-stages:
+    format: null
+run:
+    begin_stage: process-media
+    end_stage: update-vcs-config
+```
+
+### Subcommands
+
+#### `run`
+
+The `run` subcommand is used to execute one or more IUF stages. The `-b`, `-e`, `-r` and `-s` arguments can be specified to limit the stages executed. If none of those arguments are specified, `iuf run` will execute all stages
+in order. If an activity identifier is not provided via `-a`, a new activity will be created automatically.
+
+The following arguments may be specified when invoking `iuf run`:
+
+```bash
+usage: iuf run [-h] [-b BEGIN_STAGE] [-e END_STAGE] [-r RUN_STAGES [RUN_STAGES ...]]
+               [-s SKIP_STAGES [SKIP_STAGES ...]] [-F FORCE]
+               [-bc BOOTPREP_CONFIG_MANAGED [BOOTPREP_CONFIG_MANAGED ...]]
+               [-bm BOOTPREP_CONFIG_MANAGEMENT [BOOTPREP_CONFIG_MANAGEMENT ...]]
+               [-bpcd BOOTPREP_CONFIG_DIR] [-rv RECIPE_VARS] [-sv SITE_VARS]
+               [-um UPDATE_METHOD_MANAGEMENT] [-uc UPDATE_METHOD_MANAGED]
+               [--limit-managed-nodes LIMIT_MANAGED_NODES [LIMIT_MANAGED_NODES ...]]
+               [--limit-management-nodes LIMIT_MANAGEMENT_NODES [LIMIT_MANAGEMENT_NODES ...]]
+               [-mrp MASK_RECIPE_PRODS [MASK_RECIPE_PRODS ...]]
+
+options:
```

Comment:
This should be updated once development settles down.

Reply:
Updated this to most current output.  We can do it again as needed if/when other details change.